### PR TITLE
Add buildSrc gradle project

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
This will solve some things:
- No need to paste functions inside of build.gradle.kts scripts, making those more readable
- Sharing dependency versions, update them in a single place
- Slightly improved build times (barely noticable)

Not a lot of code was added, it was mostly moving things around. Some new code in `Dependencies.kt` though.

### CODE REVIEW

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * Add given project as implementation
+ */
+fun DependencyHandler.implementationProject(path: String) {
+	add("implementation", project(path))
+}
+
+/**
+ * Add the Kotlin standard library as implementation
+ */
+fun DependencyHandler.implementationKotlinStdlib() = kotlin("stdlib")
```
Comment (Maxr1998) [inline buildSrc/src/main/kotlin/Dependencies.kt:44]: If I'm not mistaken, this is missing a surrounding `implementation()`

Comment (nielsvanvelzen) [reply buildSrc/src/main/kotlin/Dependencies.kt:44]: Good catch!

File: buildSrc/src/main/kotlin/Semver.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * 0.0.0 -> 0
+ * 1.1.1 -> 10101
+ * 0.7.0 -> 700
+ * 99.99.99 -> 999999
+ */
+fun getVersionCode(semver: String): Int? {
+	val parts = semver.splitToSequence('.')
+		.take(3)
+		.map { it.toIntOrNull() }
+		.filterNotNull()
```
Comment (Maxr1998) [inline buildSrc/src/main/kotlin/Semver.kt:14]: I think you could use `mapNotNull` here.

Comment (nielsvanvelzen) [reply buildSrc/src/main/kotlin/Semver.kt:14]: Yup, IDE didn't hint it and I didn't think about it when writing the function, changed it now.

**Pull request merged**